location/lbs/activity/audioset/dataprocessing/audio_processing.py

```diff
@@ -0,0 +1,577 @@
+            temp_path = ogg_path
+        else:
+            if length == 0:
+                shutil.rmtree(tmp_path)
+            return None
+        sliced = video[start_time * 1000: end_time * 1000]
+        sliced.export(wav_path, format='wav')
+        os.remove(temp_path)
+        if length <= 1:
+            shutil.rmtree(tmp_path)
+        logging.info('chopped_audio')
+
+
+def extract_feature(dest_dir, video_id, feature, scaled):
+    """Extracts a feature from a specific audio file given a video_id.
+
+   Given a specific of video_id, it extracts features using the Librosa libray
+   from the corresponding audio file and stores the features in a dictionary
+   with a video_id, dictionary key, value pair, with the dictionary value being
+   comprised of feature name, feature list key, value pairs. Using Librosa it
+   extracts one of the following features: chroma_stft, chroma_cqt, chroma_cens,
+   melspectogram, mfcc, rms, spectral_centroid, spectral_bandwidth,
+   spectral_contrast, spectral_flatness, spectral_rolloff, poly_features,
+   tonnetz, zero_crossing_rate
+
+   Args:
+       dest_dir: Path to the parent directory where the downloaded videos are
+           stored.
+       video_id: The video_id of specific audio file from which features are to
+           be extracted.
+       feature: A dictionary with video_id, dictionary pairs with the
+           dictionary values being feature name, feature list key, value pairs.
+       scaled: whether the extracted feature should be averaged
+
+   Returns:
+       A boolean of whether the extracting features was successful
+
+   Raises: ValueError: A feature not supported by Librosa has been inputted.
+   """
+    path = join(dest_dir, 'yt_videos', 'sliced_' + video_id + '.wav')
+    if not isfile(path):
+        return None
+    try:
+        # get audio (y) and sampling rate (sr)
+        y, sr = librosa.load(dest_dir + '/yt_videos/sliced_'
+                             + video_id + '.wav')
+    except ValueError as error:
+        logging.error(error)
+        return False
+    switcher = {
+        'chroma_stft': librosa.feature.chroma_stft,
+        'chroma_cqt': librosa.feature.chroma_cqt,
+        'chroma_cens': librosa.feature.chroma_cens,
+        'melspectrogram': librosa.feature.melspectrogram,
+        'mfcc': librosa.feature.mfcc,
+        'rms': librosa.feature.rms,
+        'spectral_centroid': librosa.feature.spectral_centroid,
+        'spectral_bandwidth': librosa.feature.spectral_bandwidth,
+        'spectral_contrast': librosa.feature.spectral_contrast,
+        'spectral_flatness': librosa.feature.spectral_flatness,
+        'spectral_rolloff': librosa.feature.spectral_rolloff,
+        'poly_features': librosa.feature.poly_features,
+        'tonnetz': librosa.feature.tonnetz,
+        'zero_crossing_rate': librosa.feature.zero_crossing_rate
+    }
+    odd_ones = {'rms', 'spectral_flatness'}
+    func = switcher.get(feature)
```

Comment:
Instead of `func`, consider `feature_extraction_func`

Reply:
Done.